tests/pos/typeclass-encoding2.scala

```diff
@@ -81,22 +81,22 @@ object semiGroups {
     type Impl[T] = Extension[T, Monoid] with MonoidCommon
   }
 
-  implicit object IntOps extends Extension[Int, Monoid] with MonoidCommon { self =>
+  implicit object IntOps extends Extension[Int, Monoid] with MonoidCommon {
     type This = Int
     type Instance = Monoid
     def unit: Int = 0
     def inject($this: Int) = new Monoid {
-      val common: self.type = self
+      val common: IntOps.this.type = IntOps.this
```

Comment:
I think this could be `: IntOps.type = IntOps`

Reply:
That's the same, but less regular. I am after an encoding that works the same for classes and objects. Unfortunately, with self types for objects dropped, we only have qualified this to avoid an irregularity.